app/models/test_track/notify_assignment_job.rb

```diff
@@ -25,9 +25,30 @@ def perform
 
   private
 
-  def track
+  def maybe_track
     return "failure" unless TestTrack.enabled?
+    return "success" if skip_analytics_event?
     result = TestTrack.analytics.track(TestTrack::AnalyticsEvent.new(visitor_id, assignment))
     result ? "success" : "failure"
   end
+
+  def skip_analytics_event?
+    assignment.feature_gate? && skip_experience_event?
+  end
+
+  def skip_experience_event?
+    skip_all_experience_events? || !sample_event?
+  end
+
+  def skip_all_experience_events?
+    experience_sampling_weight.zero?
+  end
+
+  def experience_sampling_weight
+    @experience_sampling_weight ||= TestTrack::Remote::SplitRegistry.to_hash['experience_sampling_weight']
```

Comment:
to @samandmoore's point on object envy, i think this is another case where an object would provide a nicer interface.

Reply:
Yeah, `SplitRegistry.experience_sampling_weight` is probably the right place to put that method unless we want to uncork a big refactor. No new object needed.

Reply:
done